lib/enum/nested.rb

```diff
@@ -18,21 +18,28 @@ def initialize(people_array)
   end
 
   def ages_sum
+    @people.reduce(0){|sum, key| sum += key.age}
   end
 
   def average_age
+    @people.reduce(0){|sum, key| sum += key.age} / @people.count
   end
 
   def total_years_programming_experience_for_all_languages
+    @people.reduce(0) {|sum, key|
+      sum += key.years_language_experience.reduce(0) {|sum, language| sum += language[1]}}
   end
 
   def favorite_food_frequency
+    @people.map { |person| person.favorite_foods}.flatten.each_with_object(Hash.new(0)){ |key, value| value[key] += 1 }
```

Comment:
each_with_object takes two parameters, but they are NOT `|key, value|`. Instead, they are `|key, new_hash|` where new_hash is the hash you created with Hash.new(0). It doesn't make sense to say `value[key]`, but it does make sense to say `new_hash[key]`.